### PR TITLE
[Backend] Login component 

### PR DESCRIPTION
**Pull request by okasato**
## Implement backend part
  - set up database
  - write migration file for users table
  - add authentication part using passport.js in app.js
  - add some endpoints for login and register part in app.js

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -3,14 +3,51 @@ const path = require('path');
...
+    .findById(id)
+    .then(user => done(null, user))
+    .catch(err => done(err));
+});
+
 // view engine setup
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'jade');
+// app.set('views', path.join(__dirname, 'views'));
+// app.set('view engine', 'jade');
```
Comment (ChristianSchmidt1981) [inline server/app.js]: did you need this or can you delete this 2 lines?

File: server/app.js [hunk 2]
```diff
@@ -3,14 +3,51 @@ const path = require('path');
...
 
 // uncomment after placing your favicon in /public
 // app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
```
Comment (ChristianSchmidt1981) [inline server/app.js]: this 2 lines to

File: server/db/users/addUser.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module.exports = knex => {
+  return async params => {
+    const user = params;
+    try {
+      const userInfo = await knex('users').insert({
```
Comment (ChristianSchmidt1981) [inline server/db/users/addUser.js]: return directly and don't use a const userInfo

File: server/db/users/findByAccount.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = knex => {
+  return async params => {
+    const account = params;
+    try {
+      const user = await knex('users').where('email', params);
```
Comment (ChristianSchmidt1981) [inline server/db/users/findByAccount.js]: remove the const user and return directly the value

File: server/db/users/findById.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module.exports = knex => {
+  return async params => {
+    const account = params;
+    try {
+      const id = await knex('users').where('id', params);
```
Comment (ChristianSchmidt1981) [inline server/db/users/findById.js]: remove the const id and return directly the value

**Review by ChristianSchmidt1981 — APPROVED** [review]
please make the changes + answer please my questions

File: server/app.js (hunk 1 above)
Comment (okasato) [reply server/app.js]: Actually I want to delete these lines.

**Pull request merged**